scanner/cmd/scannerctl/main.go

```diff
@@ -91,23 +91,26 @@ func rootCmd(ctx context.Context) *cobra.Command {
 		}
 		// Set options for the gRPC connection.
 		opts := []client.Option{
-			client.WithAddress(*address),
-			client.WithServerName(*serverName),
+			client.WithIndexerAddress(*indexerAddr),
+			client.WithIndexerServerName(*indexerServerName),
+			client.WithIndexerSubject(mtls.ScannerV4IndexerSubject),
+			client.WithMatcherAddress(*matcherAddr),
+			client.WithMatcherServerName(*matcherServerName),
+			client.WithMatcherSubject(mtls.ScannerV4MatcherSubject),
 		}
 		if *skipTLSVerify {
 			opts = append(opts, client.SkipTLSVerification)
 		}
-		if *indexerAddr != "" {
-			opts = append(opts, client.WithIndexerAddress(*indexerAddr))
-		}
-		if *matcherAddr != "" {
-			opts = append(opts, client.WithMatcherAddress(*matcherAddr))
+		if *address != "" {
+			opts = append(opts, client.WithAddress(*address))
+			*indexerAddr, *matcherAddr = *address, *address
 		}
-		if *indexerServerName != "" {
-			opts = append(opts, client.WithIndexerServerName(*indexerServerName))
+		if *serverName != "" {
+			opts = append(opts, client.WithServerName(*serverName))
```

Comment:
nit: same comment as with `--address`, now `--server-name` has precedence over the more specific names.